### PR TITLE
Fix some issues on MacOS platform

### PR DESCRIPTION
**Pull request by wenyongh**
Fix some issues on MacOS platform
- Enable libc-wasi by default
- Set target abi to "gnu" if it is not set for wamrc to avoid generating
  object file of unsupported Mach-O format
- Set `<vendor>-<sys>` info according to target abi for wamrc to support
  generating AOT file for other OSs but not current host
- Set cpu name if arch/abi/cpu are not set to avoid checking SIMD
  capability failed
- Set size level to 1 for MacOS/Windows platform to avoid relocation type
  unsupported warning
- Clear posix_memmap.c compiling warning
- Fix spec case test script issues, enable test spec cases on MacOS

Signed-off-by: Wenyong Huang <wenyong.huang@intel.com>

### CODE REVIEW

File: core/iwasm/compilation/aot_llvm.c [hunk 1]
```diff
@@ -1610,7 +1610,71 @@ aot_create_comp_context(AOTCompData *comp_data, aot_comp_option_t option)
...
+
+                if (!default_triple) {
+                    aot_set_last_error(
+                        "llvm get default target triple failed.");
+                    goto fail;
+                }
+
+                vendor_sys = strstr(default_triple, "-");
+                bh_assert(vendor_sys);
+                memcpy(default_arch, default_triple,
```
[the remark this answers is not in the document]
Comment (wenyongh) [reply core/iwasm/compilation/aot_llvm.c]: Done

File: core/iwasm/compilation/aot_llvm.c [hunk 2]
```diff
@@ -1610,7 +1610,71 @@ aot_create_comp_context(AOTCompData *comp_data, aot_comp_option_t option)
...
+                else
+                    vendor_sys = "-pc-windows-";
+            }
+            else {
+                vendor_sys = "-pc-linux-";
+            }
+
+            bh_assert(strlen(arch1) + strlen(vendor_sys) + strlen(abi)
+                      < sizeof(triple_buf));
+            memcpy(triple_buf, arch1, strlen(arch1));
```
[the remark this answers is not in the document]
Comment (wenyongh) [reply core/iwasm/compilation/aot_llvm.c]: Done

File: wamr-compiler/main.c [hunk 3]
```diff
@@ -207,6 +208,22 @@ main(int argc, char *argv[])
     if (argc == 0 || !out_file_name)
         return print_help();
 
+    if (!size_level_set) {
+        /**
+         * Set opt level to 1 by default for Windows and MacOS as
```
[the remark this answers is not in the document]
Comment (wenyongh) [reply wamr-compiler/main.c:213]: OK, done.

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Pull request merged**